app/components/autocomplete/slash_suggestion/app_slash_suggestion/app_slash_suggestion.tsx

```diff
@@ -0,0 +1,238 @@
+        let completedDraft = `/${command} `;
+        if (Platform.OS === 'ios') {
+            completedDraft = `//${command} `;
+        }
+
+        onChangeText(completedDraft);
+
+        if (Platform.OS === 'ios') {
+            // This is the second part of the hack were we replace the double / with just one
+            // after the auto correct vanished
+            setTimeout(() => {
+                onChangeText(completedDraft.replace(`//${command} `, `/${command} `));
+            });
+        }
+
+        if (!isMinimumServerVersion(Client4.getServerVersion(), 5, 24)) {
+            this.setState({
+                active: false,
+            });
+        }
+    };
+
+    completeUserSuggestion = (base: string): (username: string) => void => {
+        return () => {
+            this.completeSuggestion(base);
+        };
+    }
+
+    completeChannelMention = (base: string): (channelName: string) => void => {
+        return () => {
+            this.completeSuggestion(base);
+        };
+    }
+
+    keyExtractor = (item: ExtendedAutocompleteSuggestion): string => item.Suggestion + item.type + item.item;
+
+    renderItem = ({item}: {item: ExtendedAutocompleteSuggestion}) => {
+        switch (item.type) {
+        case COMMAND_SUGGESTION_USER:
+            return (
+                <AtMentionItem
+                    testID={`autocomplete.at_mention.item.${item.item}`}
+                    onPress={this.completeUserSuggestion(item.Complete)}
+                    userId={item.item || ''}
+                />
+            );
+        case COMMAND_SUGGESTION_CHANNEL:
+            return (
+                <ChannelMentionItem
+                    channelId={item.item || ''}
+                    onPress={this.completeChannelMention(item.Complete)}
+                />
+            );
+        default:
+            return (
+                <SlashSuggestionItem
+                    description={item.Description}
+                    hint={item.Hint}
+                    onPress={this.completeSuggestion}
+                    theme={this.props.theme}
+                    suggestion={item.Suggestion}
+                    complete={item.Complete}
+                    icon={item.IconData}
+                />
+            );
+        }
+    }
+
+    render() {
+        const {maxListHeight, theme, nestedScrollEnabled} = this.props;
+
+        if (!this.state.active) {
+            // If we are not in an active state return null so nothing is rendered
+            // other components are not blocked.
+            return null;
+        }
+
+        const style = getStyleFromTheme(theme);
+
+        return (
+            <FlatList
+                testID='app_slash_suggestion.list'
+                keyboardShouldPersistTaps='always'
+                style={[style.listView, {maxHeight: maxListHeight}]}
+                extraData={this.state}
+                data={this.state.dataSource}
```

Comment:
How does `extraData` work here? Should it be an array of data rather than the component's state?

Reply:
I didn't dive much on the behavior. Mainly copy/pasted from `slash_suggestion.tsx`.

Reply:
I didn't dive much on the behavior. Mainly copy/pasted from `slash_suggestion.tsx`.

Reply:
Looks like it's meant for supplying an indicator that the list items may need to update based on an arbitrary state change https://reactnative.dev/docs/flatlist#extradata. Here we are saying, "if this component's state changes at all (because we are passing the immutable `this.state` itself), call `renderItem` again for each item."

Reply:
if you look at the postlist component, you'll see how it is used.

extraData here is doing nothing as there is nothing extra that you need for the list to re-render that is not related to state